sim/priest/shadow/spriest.go

```diff
@@ -0,0 +1,246 @@
+		//  Shadow Weaving, Vampiric Touch, and Misery
+		spriest.Character.AddPermanentAura(func(sim *core.Simulation) core.Aura {
+			return core.Aura{
+				ID:   ShadowWeaverAuraID,
+				Name: "Shadow Weaver",
+				OnPeriodicDamage: func(sim *core.Simulation, spellCast *core.SpellCast, spellEffect *core.SpellEffect, tickDamage float64) float64 {
+					// Handle talents here for now.
+					// 'always on' type effects could maybe go in as a DotEffect.DamageMultiplier that is added on each tick (separate)
+					if spellCast.SpellSchool == stats.ShadowSpellPower {
+						tickDamage *= 1 + float64(spriest.Talents.Darkness)*0.02
+						if spriest.Talents.Shadowform {
+							tickDamage *= 1.15
+						}
+					}
+					if tickDamage > 0 && spriest.VTSpell.DotInput.IsTicking(sim) {
+						s := stats.Stats{stats.Mana: tickDamage * 0.05}
+						if sim.Log != nil {
+							sim.Log("VT Regenerated %0f mana.\n", s[stats.Mana])
+						}
+						spriest.Party.AddStats(s)
+					}
+					return tickDamage
+				},
+				OnSpellHit: func(sim *core.Simulation, spellCast *core.SpellCast, spellEffect *core.SpellEffect) {
+					addShadowWeaving(sim, spellEffect)
+					if spellEffect.Damage > 0 && spriest.VTSpell.DotInput.IsTicking(sim) {
+						s := stats.Stats{stats.Mana: spellEffect.Damage * 0.05}
+						if sim.Log != nil {
+							sim.Log("VT Regenerated %0.1f mana.\n", s[stats.Mana])
+						}
+						spriest.Party.AddStats(s)
+					}
+
+					if spellCast.ActionID.SpellID == priest.SpellIDSWP || spellCast.ActionID.SpellID == priest.SpellIDVT || spellCast.ActionID.SpellID == priest.SpellIDMF {
+						spellEffect.Target.ReplaceAura(sim, core.Aura{
+							ID:      core.MiseryDebuffID,
+							Expires: sim.CurrentTime + misDur,
+							Name:    "Misery",
+							OnBeforeSpellHit: func(sim *core.Simulation, spellCast *core.SpellCast, spellEffect *core.SpellEffect) {
+								spellEffect.DamageMultiplier *= 1.05
+							},
+							OnPeriodicDamage: func(sim *core.Simulation, spellCast *core.SpellCast, spellEffect *core.SpellEffect, tickDamage float64) float64 {
+								return tickDamage * 1.05
+							},
+						})
+					}
+				},
+			}
+		})
+	}
+
+	return spriest
+}
+
+type ShadowPriest struct {
+	priest.Priest
+
+	swStacks float64
+
+	rotation proto.ShadowPriest_Rotation
+}
+
+func (spriest *ShadowPriest) GetPriest() *priest.Priest {
+	return &spriest.Priest
+}
+
+func (spriest *ShadowPriest) Reset(sim *core.Simulation) {
+	spriest.Priest.Reset(sim)
+	spriest.swStacks = 0
+}
+
+func (spriest *ShadowPriest) Act(sim *core.Simulation) time.Duration {
+	// Activate shared behaviors
+	target := sim.GetPrimaryTarget()
+	var spell *core.SimpleSpell
+	var wait time.Duration
+
+	timeForDots := sim.Duration-sim.CurrentTime > time.Second*12
+	if spriest.UseShadowfiend &&
+		spriest.CurrentMana()/spriest.MaxMana() < 0.5 &&
+		spriest.GetRemainingCD(priest.ShadowfiendCD, sim.CurrentTime) == 0 &&
+		!spriest.ShadowfiendSpell.DotInput.IsTicking(sim) {
```

Comment:
I think checking for shadowfiend cooldown will prevent this from happening, even without checking for the dot.

Reply:
I originally had it that way... but it bothered me to not check if its ticking... I guess I can remove it :)